### ISSUE TITLE
Is there a way to include callbacks in between lines ? if not, how can someone gracefully implement it ?

### ISSUE THREAD

**Issue by aymen157** [ocornut/imgui#8017]
### Version/Branch of Dear ImGui:

1.9

### Back-ends:

imgui_impl_opengl

### Compiler, OS:

Windows 11

### Details:

for example (c#)
```
ImGui.Text("Hello");
ImGui.CallbackCall("textStyler");
ImGui.Text("Hello with Injected Style");
ImGui.CallbackCall("afterStyler");
```

later in any part of the project we can do
```
ImGui.PushCallback("textStyler", pointerToSomeCoolStyleMethod);
ImGui.PushCallback("afterStyler", pointerToSomeDoubleClickHandler);
```

like this anyone can extend style without having to modify the source code of the app. pushing styles, or even handling double clicks etc..
i'm wondering if there's something like this already i missed ?

**Comment by aymen157** [comment]
> Why not doing: ....

im currently doing that. and at high scale, it becomes terrible. questions start to popup so frequently like "where did i put this?" or "should i define this callback in this or in this" etc... these feel ok at lower scale but as things start to grow it becomes a headache. now imagine that with not your software only, but others libs too.

you could think of it of openpop and if(popup) ... why not define a global static bool for that popup if its open etc.. ?
but this is much worse if you want much extensibility (as it becomes much more frequent) (you can see in that example, a text call took 2 callbacks)

that aside i think Imgui.Callback() is prettier :) than ImGuiExtensionNum1Part1Ep2.Callback()

**Comment by ocornut** [comment]
I don't really understand what you are aiming at to be honest. You seem to be pushing against dear imgui's design.

Can you share actual code you are using?

Again, if you like the sample code you suggested you are feel to implement it exactly as you wrote it.. you don't need to modify imgui for that. But I am thinking your design is fishy here.

**Comment by ocornut** [comment]
I'm afraid that we have a misunderstanding here. Nothing here actually has anything to do with Dear ImGui and its design.

If you want your code to be pluggable by third-party code, it's up to you to come up with a design that works for whatever you want to make. And there are many solutions and perhaps people here or elsewhere will help you. But I personally cannot. Dear ImGui display the things you are asking it to display.

When you ask about this in programming forums you shouldn't be thinking about "callbacks" or "classes" or any predetermined solutions that will hinder possible answers. Instead, try to explain your problem with concrete use cases. As you dig into this you will understand that your problem has nothing to do with Dear ImGui itself.

**Comment by GamingMinds-DanielC** [comment]
If you want callback support in your windows (or in custom widgets), there is nothing preventing you from adding it in. But as Omar said, that has nothing to do with ImGui directly.

If you are struggling with the right approach, you can use or get inspiration from other libraries that are designed to provide exactly that, like f.e. this one:
https://github.com/libsigcplusplus/libsigcplusplus

**Comment by aymen157** [comment]
No. having a global state machine for callbacks that will be used to draw is not a bad idea at all. I'm pretty sure it will be scattered across so many projects since its reduces so much friction at higher scale. even if you can't see it. i do. it's whatever though 👍.
